webui/src/pages/repositories/repository/settings/general.jsx

```diff
@@ -105,10 +105,11 @@ const SettingsContainer = () => {
                 repo={repo}
                 onCancel={() => { setShowDeleteModal(false) }}
                 onSubmit={() => {
-                    repositories.delete(repo.id).catch(err => {
-                        setDeletionError(err)
-                    }).then(() => {
+                    repositories.delete(repo.id).then(() => {
```

Comment:
This change seems unrelated.  Can you explain what it does?

Reply:
Without this change then is called in case of catch
With the previous implementation, if the user isn't authorized he gets an error when trying to delete the repository but the `then` is called anyway and the user is redirected to the repositories page (which in some cases the user doesn't have permissions to view that page)